rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -121,15 +126,23 @@ def run_check(cls, results_dir, **kwargs):
     @classmethod
     def parse_abi_logs(cls, reports):
         """
-        Parse abipkgdiff logs
-        :param reports: dictionary with paths to the logs of the produced (sub)packages
-        :return: returns dict of packages names with its abipkgdiff changes
+        Parse abipkgdiff logs.
+
+        Args:
+            reports(dict): Dictionary with paths to the logs of the produced (sub)packages.
+
+        Returns:
+            dict: Returns dict of packages names with its abipkgdiff changes.
```

Comment:
This is how it should look like:

```
"""
Parses summary information from abipkgdiff logs.

Sets abi_changes attribute if there are any ABI changes found.

Args:
    reports(dict): Dictionary mapping package names to abipkgdiff return codes.

Returns:
    dict: Dictionary mapping package names to a list of summary information for each shared object.

    For example:

    {
        'libtiff': [
            {
                'filename': 'libtiff.so.5.2.5',
                'Functions changes summary': {
                    'Added': {
                        'count': '8',
                        'what': 'Added',
                        'filtered_out': None,
                    }
                }
            },
            {
                'filename': 'libtiff.so.5.2.5',
                'Variables changes summary': {},
            },
        ],
        'libtiff-tools': [],
    }
"""
```

Also, looking at the result, it appears to me that it could be structured in a better way.

Reply:
New structure of the analyzes output.
```
{'pstoedit': 
    {
    'libpstoedit.so.0.0.0': 
        {
        'Functions changes summary': 
            {
                'Removed': {'count': '1', 'what': 'Removed', 'filtered_out': None}, 
                'Added': {'count': '3', 'what': 'Added', 'filtered_out': None}, 
                'Changed': {'count': '2', 'what': 'Changed', 'filtered_out': '19'}
            }, 
        'Variables changes summary': 
            {'Changed': {'count': '1', 'what': 'Changed', 'filtered_out': '1'}
        }
    }, 
    'libp2edrvstd.so.0.0.0': 
        {
        'Functions changes summary': 
            {'Changed': {'count': '1', 'what': 'Changed', 'filtered_out': '63'}
        }, 
        'Variables changes summary': {}
    }, 
    'libp2edrvlplot.so.0.0.0': 
        {
        'Functions changes summary': 
                {'Changed': {'count': '1', 'what': 'Changed', 'filtered_out': '1'}}, 
         'Variables changes summary': {}
     }, 
    'libp2edrvwmf.so.0.0.0': 
        {
        'Functions changes summary': 
                {'Changed': {'count': '1', 'what': 'Changed', 'filtered_out': '1'}}, 
        'Variables changes summary': {}
        }
    }
}
```

Reply:
By deleting empty dictionary, we miss empty `pstoedit-devel.txt` file. I suggest we leave it this way, as it indeed does not contain any details...

Reply:
I now changed it to directly assign the output:
`pkgs[pkg] = parse_changes(f.readlines())`
as the structure has been changed.

After the change, only nonempty pairs `{package: abi_changes}` are added to the `AbiPkgDiff` dictionary output. Therefore only packages with changes are mentioned in the `report.txt` file. Do you think it is worth mentioning also packages without changes? I suggest leaving it the way it is.

So the output is:
```
Details in rebase-helper-results/checkers/abipkgdiff:
 - pstoedit.txt
```
 and not
```
Details in rebase-helper-results/checkers/abipkgdiff:
 - pstoedit.txt
 - pstoedit-devel.txt(empty)
```

Reply:
I think this is fine. It would be good to also delete those empty abipkgdiff logs, but that's out of scope of this PR I think.